### PR TITLE
login dialog

### PR DESCRIPTION
**Pull request by justynasadlak**

### CODE REVIEW

File: src/app/modules/user/login-dialog/login-dialog.component.html [hunk 1]
```diff
@@ -1,25 +1,23 @@
...
+  </mat-form-field>
+  <mat-form-field>
+    <input matInput type="password" placeholder="password" formControlName="password">
+  </mat-form-field>
+  <mat-form-field *ngIf="isRegisterView">
+    <input matInput type="password" placeholder="repeat password" formControlName="repeatPassword">
+  </mat-form-field>
+  <mat-checkbox formControlName="rememberMe">remember me</mat-checkbox>
+  <button mat-flat-button color="primary" type="submit">
+    {{ getButtonName() }}
```
Comment (figuratively) [inline src/app/modules/user/login-dialog/login-dialog.component.html]: Do not call functions in HTML. You should use a variable.

File: src/styles.scss [hunk 2]
```diff
@@ -9,3 +9,19 @@ body {
...
+.login__wrapper {
+  height: 50vh;
+}
+
+.register__wrapper, .login__wrapper {
+  transition: all 0.5s ease-in-out;
+}
+
+.mat-dialog-container {
+  overflow: hidden !important;
```
Comment (figuratively) [inline src/styles.scss]: I think that overflow:hidden could work without !important if you added it to the class you appended to the dialog.

File: src/app/containers/hotel-details/hotel-details.component.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+  private bookingData: Booking;
+  private startDate;
+  private endDate;
+  private username;
+  private roomsToBook = [];
+  private successBooking: Booking;
+
+  public constructor(private route: ActivatedRoute,
+                     private _location: Location,
```
Comment (figuratively) [inline src/app/containers/hotel-details/hotel-details.component.ts]: Do not use underscore.

File: src/app/app.module.ts [hunk 4]
```diff
@@ -16,6 +16,8 @@ import { SearchHotelsListComponent } from './components/search-hotels-list/searc
 import { SearchBoxComponent } from './components/search-box/search-box.component';
 import { UserBookingsComponent } from './modules/user/user-bookings/user-bookings.component';
 import {RouterModule} from '@angular/router';
+import {HotelDetailsComponent} from './containers/hotel-details/hotel-details.component';
+import { HotelDetailsViewComponent } from './components/hotel-details-view/hotel-details-view.component';
```
Comment (Shaelex) [inline src/app/app.module.ts]: Formatting

File: src/app/components/hotel-details-view/hotel-details-view.component.html [hunk 5]
```diff
@@ -0,0 +1,22 @@
+<mat-card>
+  <button mat-raised-button color="primary" (click)="onBack()">back</button>
+  <mat-card-header>
+    <mat-card-title>
+      {{ name}}
```
Comment (Shaelex) [inline src/app/components/hotel-details-view/hotel-details-view.component.html]: Formatting

File: src/app/components/hotel-details-view/hotel-details-view.component.html [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    </mat-card-title>
+    <mat-card-subtitle>{{ location}}</mat-card-subtitle>
```
Comment (Shaelex) [inline src/app/components/hotel-details-view/hotel-details-view.component.html]: as above

File: src/app/components/hotel-details-view/hotel-details-view.component.ts [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+  @Input()
+  rooms: Room[];
+
+  @Input()
+  isDisabled: boolean;
+
+  @Output()
+  book: EventEmitter<string> = new EventEmitter<string>();
+
+  constructor() {
```
Comment (Shaelex) [inline src/app/components/hotel-details-view/hotel-details-view.component.ts]: An empty constructor can be deleted.

File: src/app/components/hotel-details-view/hotel-details-view.component.ts [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+  }
+
+  ngOnInit(): void {
```
Comment (Shaelex) [inline src/app/components/hotel-details-view/hotel-details-view.component.ts]: An empty `ngOnInit` can be deleted.

File: src/app/containers/hotel-details/hotel-details.component.ts [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+  name: string;
+  location: string;
+  facilities;
+  rooms;
+  isDisabled = false;
+
+  private bookingData: Booking;
+  private startDate;
+  private endDate;
+  private username;
```
Comment (Shaelex) [inline src/app/containers/hotel-details/hotel-details.component.ts]: Missing types.

File: src/app/containers/hotel-details/hotel-details.component.ts [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+  templateUrl: './hotel-details.component.html',
+  styleUrls: ['./hotel-details.component.scss']
+})
+export class HotelDetailsComponent implements OnInit {
+
+  id: string;
+  name: string;
+  location: string;
+  facilities;
+  rooms;
```
Comment (Shaelex) [inline src/app/containers/hotel-details/hotel-details.component.ts]: Missing types.

File: src/app/containers/hotel-details/hotel-details.component.ts [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+  private roomsToBook = [];
+  private successBooking: Booking;
+
+  public constructor(private route: ActivatedRoute,
+                     private _location: Location,
+                     private store: Store,
+                     private hotelService: HotelService,
+                     private bookingService: BookingService,
+                     private router: Router) {
+    this.route.queryParams.subscribe(params => {
```
Comment (Shaelex) [inline src/app/containers/hotel-details/hotel-details.component.ts]: Should be placed in `ngOnInit`

File: src/app/containers/hotel-details/hotel-details.component.ts [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+      this.id = params.id;
+      this.name = params.name;
+      this.location = params.location;
+      this.facilities = JSON.parse(params.facilities);
+      this.rooms = JSON.parse(params.rooms);
+    });
+  }
+
+  ngOnInit(): void {
+    console.log(this.rooms);
```
Comment (Shaelex) [inline src/app/containers/hotel-details/hotel-details.component.ts]: Console.logs aren't allowed in production code.

File: src/app/containers/hotel-details/hotel-details.component.ts [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+      this.getStartDate();
+      this.getEndDate();
+      this.getUsername();
+      this.bookingData = {
+        user: this.username,
+        startDate: this.startDate,
+        endDate: this.endDate,
+        rooms: this.roomsToBook
+      };
+      console.log(this.bookingData);
```
Comment (Shaelex) [inline src/app/containers/hotel-details/hotel-details.component.ts]: No console logs

File: src/app/containers/hotel-details/hotel-details.component.ts [hunk 14]
```diff
@@ -0,0 +1,95 @@
...
+    console.log(this.rooms);
+  }
+
+  onBack(): void {
+    this._location.back();
+  }
+
+  onBook(roomId: string): Observable<Booking> {
+    this.isDisabled = true;
+    this.hotelService.getRoom(roomId).subscribe(room => {
```
Comment (Shaelex) [inline src/app/containers/hotel-details/hotel-details.component.ts]: Make sure if this autocompletes.

File: src/app/containers/hotel-details/hotel-details.component.ts [hunk 15]
```diff
@@ -0,0 +1,95 @@
...
+      this.getEndDate();
+      this.getUsername();
+      this.bookingData = {
+        user: this.username,
+        startDate: this.startDate,
+        endDate: this.endDate,
+        rooms: this.roomsToBook
+      };
+      console.log(this.bookingData);
+      this.bookingService.addBooking(this.bookingData).subscribe(val => {
```
Comment (Shaelex) [inline src/app/containers/hotel-details/hotel-details.component.ts]: Make sure if this autocompletes.

File: src/app/containers/hotel-details/hotel-details.component.ts [hunk 16]
```diff
@@ -0,0 +1,95 @@
...
+          alert(error1);
+          this.isDisabled = false;
+        });
+
+    });
+    return of(this.successBooking);
+  }
+
+  private getStartDate(): void {
+    this.store.select('startDate').pipe(tap(date =>
```
Comment (Shaelex) [inline src/app/containers/hotel-details/hotel-details.component.ts]: Is subscribe needed? What about memory leaks?

File: src/app/containers/landing-page/landing-page.component.ts [hunk 17]
```diff
@@ -88,7 +95,14 @@ export class LandingPageComponent implements OnInit {
         rooms: this.rooms
       };
       console.log(this.bookingData);
-      this.bookingService.addBooking(this.bookingData).subscribe();
+      this.bookingService.addBooking(this.bookingData).subscribe(x => {
```
Comment (Shaelex) [inline src/app/containers/landing-page/landing-page.component.ts]: Make sure if this autocompletes.

File: src/app/services/booking.service.ts [hunk 18]
```diff
@@ -19,13 +16,12 @@ export class BookingService {
...
 
 
-
   getAllBookings(): Observable<Booking[]> {
     return this.http.get<Booking[]>(this.url + '/bookings').pipe(delay(2000), tap(response => this.store.set('bookings', response)));
   }
 
   addBooking(booking: Booking): Observable<Booking> {
-    return this.http.post<Booking>(this.url + '/bookings', booking);
+    return this.http.post<Booking>(this.url + '/bookings', booking).pipe(delay(2000));
```
Comment (Shaelex) [inline src/app/services/booking.service.ts]: why delay?

File: src/styles.scss (hunk 2 above)
Comment (Shaelex) [reply src/styles.scss]: Try to make it w/o !important.

**Pull request merged**